oauthproxy.go

```diff
@@ -684,7 +684,12 @@ func (p *OAuthProxy) Authenticate(rw http.ResponseWriter, req *http.Request) int
 		if session.Email != "" {
 			rw.Header().Set("X-Auth-Request-Email", session.Email)
 		}
+
+		if p.PassAccessToken && session.AccessToken != "" {
+			rw.Header().Set("X-Auth-Request-Access-Token", session.AccessToken)
+		}
 	}
```

Comment:
Had to do this because setting the command line arg --pass-access-token=true and setting 

```
auth_request_set $access_token $upstream_http_x_forwarded_access_token;
```
and
```
proxy_set_header X-Forwarded-Access-Token $access_token;
```

in Nginx didn't set the X-Forwarded-Access-Token header.

Reply:
When sending this upstream, probably want to do this as two PRs. It's worth thinking about how you're going to do this.

I'd suggest raising a separate PR for each here and using the same branches to submit them upstream.